File: arrays/average-waiting-time/average-waiting-time.js
/**
 * @param {number[][]} customers
 * @return {number}
 */
var averageWaitingTime = function (customers) {
  let chef = 0;
  let timeWaited = 0;

  for (let i = 0; i < customers.length; i++) {
    chef = Math.max(chef, customers[i][0]) + customers[i][1];

    timeWaited += chef - customers[i][0];
  }

  return timeWaited / customers.length;
};

File: arrays/average-waiting-time/average-waiting-time.go
func averageWaitingTime(customers [][]int) float64 {
	chef := 0.0
	timeWaited := 0.0

	for i := 0; i < len(customers); i++ {
		chef = (math.Max(chef, float64(customers[i][0])) + float64(customers[i][1]))

		timeWaited += (chef - float64(customers[i][0]))
	}

	return timeWaited / float64(len(customers))
}
